tools/clang/test/HLSLFileCheck/passes/hl/gvn-eliminate-region/gvn-eliminate-region.hlsl

```diff
@@ -0,0 +1,45 @@
+// RUN: %dxc -T ps_6_0 -E main /opt-disable gvn %s | FileCheck %s
```

Comment:
It would be good to add a few more tests here. 

1. Cases where the syntax is not identical, but is equivalent (e.g. foo=a+b, bar=b+a).
2. Cases where there are memory operations. Do we expect these to be eliminated?

It would also be good to add some negative tests. Ideally anywhere we have a `return false` in the `CompareOrBuild` we would have a test that exercises that condition.

Reply:
Added some new tests. Interestingly, simple commutative expressions work fine, but we can't really handle expressions that are identical from associativity. Maybe there's more opportunities there that we can explore later.